pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -0,0 +1,33 @@
+import signal
+
+
+class SignalConnector:
+    """
+    Takes care of registering and restoring signal handlers for the
+    :class:`~pytorch_lightning.trainer.trainer.Trainer`. By default, it handles
+    SIGTERM, SIGINT and SIGSEGV by raising KeyboardInterrupt and letting Trainer do graceful shutdown.
+    """
+
+    def __init__(self, trainer):
+        self.trainer = trainer
```

Comment:
Question: Do we really want this to hold a hard reference? shouldn't we use a weakref here instead? Usually this class should not prevent the trainer from being destroyed and this class will be destroyed if the trainer holding it will be destroyed. This might lead to circular references.

Reply:
I followed @williamFalcon 's pattern of passing the trainer to every connector. I think I don't actually need the reference in this connector, so i could remove it.

Reply:
Then this is maybe a more general solution. I guess a weakref would be the better option here.

cc @Borda @williamFalcon

Reply:
ok, so you suggest to do something like this:
```python
t = weakref.ref(trainer)
c1 = Connector1(t)
c2 = Connector2(t)
...
```
?

Reply:
yes or you create the weakref inside the connector. Doesn't matter but would probably be easier to read

Reply:
cool. so at some point the connectors could probably benefit from a super class. we could take care of the weakref for trainer in a superclass that every connector inherits from?